### PR TITLE
[SPARK-13042][MLlib] Add Specificity (= true negative rate) to MulticlassMetrics.scala

### PR DESCRIPTION
**Pull request by dmarcous**
Add specificity calculations

pecificity (SPC) or true negative rate
SPC = TN / N = TN / (TN+FP) 
False Positive Rate (FPR)
FPR = FP / N = FP / (FP + TN) = 1-SPC

SPC = 1 - FPR

### CODE REVIEW

**Comment by srowen** [comment]
There are a number of things wrong with this; please read https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark first. I don't think this is useful enough to include, and your tags are wrong anyway.

**Comment by srowen** [comment]
This is not related to SPARK-10945

File: mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala [hunk 1]
```diff
@@ -116,6 +116,13 @@ class MulticlassMetrics @Since("1.1.0") (predictionAndLabels: RDD[(Double, Doubl
   def recall(label: Double): Double = tpByClass(label).toDouble / labelCountByClass(label)
 
   /**
+   * Returns specificity for a given label (category)
+   * @param label the label.
+   */
+  @Since("1.6.0")
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala]: This is still wrong. This can't have existed in 1.6.0

**Comment by mengxr** [comment]
ok to test

**Comment by dmarcous** [comment]
Removed extra whitespaces in order to pass style tests

**Comment by dmarcous** [comment]
corrected typo - duplicate rows paste

**Comment by dmarcous** [comment]
hi guys, any update? seems like its ready for your blessing

**Comment by srowen** [comment]
I don't object to someone merging this but I don't think it is worth including in the API. It is trivial to calculate.

**Comment by dmarcous** [comment]
My thinking was adding it so it would make code of people using it "cleaner" just specifying another metric by name, and not adding a calculation in user code near the other metrics.

So, if we want to merge and not include in the API does it mean I should mark the new methods as private? remove annotations so it would be undocumented? whats the best way to go from your experience?

**Comment by srowen** [comment]
I don't think that would be useful, to make a private unused API.

**Comment by srowen** [comment]
I believe we should resolve this as WontFix if there aren't other committers supporting this change

**Comment by dmarcous** [comment]
Agreed that it should be resolved.
Is there a process where I can reach out for any committer for support?
I'm still in need of this and think those extra couple of lines goes a long way in user code readability when using it from inside the class

**Comment by srowen** [comment]
Paging ... @MLnick ? Usually I find people who have changed surrounding code.

**Comment by MLnick** [comment]
I'd only say it's worth putting in if it's something that users would cross-validate against (in which case makes more sense to put it into the `ml` evaluator). But I don't think that's a common case, so unless we saw a lot of user requests I'd tend to lean against it, since as you mention it's pretty straightforward to write a small function to compute it.

**Comment by dmarcous** [comment]
Got it, thank you @MLnick and thank you @srowen for your inputs along the way, is has been helpful!
Closing the request.

**Pull request closed without merging**